newa/cli.py

```diff
@@ -194,50 +193,99 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str]) -> None:
         config = ErratumConfig.from_yaml_file(Path(os.path.expandvars(issue_config)))
 
-        # TODO: record created/existing issues. Instead of `Any`, maybe something
-        # from the Jira library would be stored. Or just a Jira ticket ID.
-        known_issues: dict[str, Any] = {}
+        # Record created/existing issues.
+        known_issues: dict[str, Issue] = {}
 
         # Iterate over issue actions. Take one, if it's not possible to finish it,
         # put it back at the end of the queue.
         issue_actions = config.issues[:]
 
         while issue_actions:
             action = issue_actions.pop(0)
+            rendered_summary = render_template(action.summary, ERRATUM=erratum_job)
+            rendered_description = render_template(action.description, ERRATUM=erratum_job)
+            rendered_assignee = render_template(action.assignee, ERRATUM=erratum_job)
 
+            # For debugging.
             print(f'* Would create a {action.type.name} issue:')
             print(f'     summary: {action.summary}')
             print(f'     summary: {action.description}')
             print()
 
-            if action.id in known_issues:
-                raise Exception(f'Issue "{action.id}" is already created!')
-
+            # Find existing issues in Jira related to erratum_job and action.id.
+            #
+            # We re looking for Jira issues such that:
+            #
+            # 1) summary matches rendered_summary (e.g. Errata respin 0) and
+            # 2) description contains erratum_job.erratum (e.g. Erratum(130704) and
+            # 3) description contains erratum_job.erratum.release (e.g. Release(RHEL-8.8.0.Z.EUS))
+            query = f""" summary ~ '{rendered_summary}' and
+                         description ~ '{erratum_job.erratum!r}' and
+                         description ~ '{erratum_job.erratum.release!r}' """
+            for jira_issue in jira.search_open_issue(query):
+                desc = jira_issue.fields.description
+
+                # We found relevant issue in Jira, it is either related to the
+                # current respin or not.
+                #
+                # 1) If it is, it is not created again - it is skipped.
+                if isinstance(desc, str) and desc.find(repr(erratum_job.erratum.builds)):
+                    known_issues[action.id] = Issue(jira_issue.key)
+                    print(f"Issue '{action.id}' is already created ({known_issues[action.id]})")
+
+                # 2) If it is not, then it is related to the previous respin(s), its fate is
+                # driven by action.on_respin - it is either closed or kept open.
+                else:
+                    if action.on_respin == "close":
+                        key = jira_issue.key
+                        jira.close_issue(key)
+                        print(f"Closed issue {key}")
+
+            # For debugging.
             if action.parent_id and action.parent_id not in known_issues:
```

Comment:
I think this is important and we should not be doing the above processing before this block.

Reply:
All fixed in the current version.